x-pack/docs/en/security/limitations.asciidoc

```diff
@@ -90,3 +90,28 @@ LDAP Groups.  For example, if a user is a member of `group_1` and `group_1` is a
 member of `group_2`, only `group_1` will be discovered. However, the
 <<active-directory-realm, Active Directory Realm>> *does* support transitive
 group membership.
+
+
+[discrete]
+[[same-user-check]]
+=== Same user check
+
+The result of <<async-search,async search>> and <<scroll-api,scroll>> can be retrieved later
+by the same user who submits the initial request. The same user check is done by comparing
+the username, authentication realm type and, for realms other than file or native, realm name.
+Two API keys are never considered as the same user regardless of their privileges.
+This logic also has a few limitations:
+
+* Two different realms can have the same name on different nodes. We do not
+recommend configuring realms in this inconsistent way. As such the same user check
+does not attempt to detect this inconsistency.
+* The same realm can also be renamed. This can cause inconsistency for same user check
+when submitting async search/scroll before the rename and retrieving the result after the rename.
+. Hence changing realm names should be handled with care
+since it can cause complications for more than just same user check.
+* The username is computed on the fly for realms backed by certain external authentication
```

Comment:
I'm not sure that "on the fly" is clear in all languages. Can we use "dynamically" instead?:
```suggestion
* The username is dynamically computed for realms backed by certain external authentication
```